editor/find_in_files.cpp

```diff
@@ -233,8 +233,9 @@ void FindInFiles::_scan_dir(String path, PackedStringArray &out_folders) {
 			break;
 		}
 
-		// Ignore special dirs (such as .git and .import)
-		if (file == "." || file == ".." || file.begins_with(".")) {
+		// Ignore special dirs (such as .git and project data directory)
+		String project_data_dir_name = ProjectSettings::get_singleton()->get_project_data_dir_name();
+		if (file == "." || file == ".." || file.begins_with(".") || file == project_data_dir_name) {
```

Comment:
Maybe we can simplify since the `begins_with(".")` check also handles `.` and `..`.
```suggestion
		if (file.begins_with(".") || file == project_data_dir_name) {
```